includes/in-person-payments/class-wc-payments-in-person-payments-receipts-service.php

```diff
@@ -84,63 +84,52 @@ private function format_line_items( array $order ) :array {
 	 * @return void
 	 * @throws \Exception Error validating settings.
 	 */
-	private function validate_settings( $settings ) {
+	private function validate_settings( array $settings ) {
 		if ( ! $settings ) {
-			throw new \Exception( 'You must provide settings information' );
+			throw new \Exception( 'You must provide settings information.' );
 		}
 
 		if ( ! array_key_exists( 'business_name', $settings ) ) {
-			throw new \Exception( 'You must provide a business name' );
+			throw new \Exception( 'You must provide a business name.' );
 		}
 
-		$this->validate_support_info( $settings['support_info'] );
-	}
-
-	/**
-	 * Validate support info
-	 *
-	 * @param  array $support_info Support info.
-	 * @return void
-	 * @throws \Exception Error validating support info.
-	 */
-	private function validate_support_info( $support_info ) {
-		if ( ! $support_info ) {
-			throw new Exception( 'You must provide support information.' );
+		if ( ! array_key_exists( 'support_info', $settings ) ) {
+			throw new \Exception( 'You must provide support information.' );
 		}
 
-		foreach ( $support_info as $key => $value ) {
+		foreach ( $settings['support_info'] as $key => $value ) {
 			if ( ! in_array( $key, [ 'address', 'email', 'phone' ], true ) ) {
-				throw new Exception( 'Invalid support information.' );
+				throw new Exception( sprintf( 'Error validating support information. Invalid key: %s', $key ) );
 			}
 		}
 	}
 
 	/**
 	 * Validate charge information
 	 *
-	 * @param  mixed $charge Charge info.
+	 * @param  array $charge Charge info.
 	 * @return void
 	 * @throws \Exception Error validating charge info.
 	 */
-	private function validate_charge( $charge ) {
+	private function validate_charge( array $charge ) {
 		if ( ! $charge ) {
-			throw new Exception( 'You must provide charge information' );
+			throw new Exception( 'You must provide charge information.' );
 		}
 
 		if ( ! array_key_exists( 'amount_captured', $charge ) ) {
-			throw new Exception( 'You must provide a captured amount' );
+			throw new Exception( 'You must provide a captured amount.' );
 		}
 
-		if ( ! array_key_exists( 'payment_method_details', $charge ) ) {
-			throw new Exception( 'You must provide payment method details' );
+		if ( ! array_key_exists( 'payment_method_details', $charge ) || ! is_array( $charge['payment_method_details'] ) || ! $charge['payment_method_details'] ) {
```

Comment:
I'm not 100% sure, but aren't we expecting `string` for these 3 fields? If so, then `is_string` would be clearer choice over `! is_array`

Reply:
`! array_key_exists( 'payment_method_details', $charge ) || ! is_array( $charge['payment_method_details'] ) || ! $charge['payment_method_details']` -> `is_empty($charge['payment_method_details'])`. Not big fan of `empty`, but here it makes sense.

Reply:
thanks for checking this.

@KarlisJ payment_method_details is an associative array, it comes from Stripe's API response

@kalessil great I'll use is_empty to check for empty array